src/platform/packages/shared/response-ops/rule_form/src/common/hooks/use_load_rule_type_aad_template_fields.ts

```diff
@@ -24,12 +25,24 @@ export interface UseLoadRuleTypeAadTemplateFieldProps {
 }
 
 export const useLoadRuleTypeAadTemplateField = (props: UseLoadRuleTypeAadTemplateFieldProps) => {
+  const ecsFlat = useRef<Record<string, any>>({});
   const { http, ruleTypeId, enabled, cacheTime } = props;
 
   const queryFn = async () => {
     if (!ruleTypeId) {
       return;
     }
+    if (isEmpty(ecsFlat.current)) {
+      // If we use import { EcsFlat } from '@elastic/ecs' then it will sometimes balloon the bundle size
+      // by about 1MB, or not, depending on how Webpack is feeling today. If you delete this dynamic import,
+      // it's possible that it won't make the bundle size explode, and then you will commit that change, and
+      // we will all continue to live our lives safe and secure, falsely believing that the danger has passed,
+      // until one day somebody makes another change to this package and, for mysterious, unknowable reasons,
+      // Webpack decides that today the bundle size shall be engorged once again.
+      // To avoid all that maybe just don't delete this dynamic import.
+      ecsFlat.current = await import('@elastic/ecs').then((ecs) => ecs.EcsFlat);
```

Comment:
There is a service to dynamically load ESC data for selected fields: https://github.com/elastic/kibana/blob/main/x-pack/platform/plugins/shared/fields_metadata/README.md

Or when rendering it's possible to use `<FieldDescription .../>` component from https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-field-utils/src/components/field_description/field_description.tsx

Reply:
will this be addressed in this PR?

Reply:
I've honestly had to switch this back and forth from dynamic import to static import like 3 times over the past couple months of making changes to `alerts-ui-shared` and the rule form package, everything has been unrelated and just moving other code around, so I haven't had time to actually examine what EcsFlat is used for here. On `main` Webpack puts it in an asnyc chunk and on this PR for some reason it changes its mind about that. Handling this better is beyond the scope of this PR, I'm just trying to keep the bundle size down so it can merge.

Reply:
Issue: https://github.com/elastic/kibana/issues/211585.